addons/binding/org.openhab.binding.opendaikin/src/main/java/org/openhab/binding/opendaikin/internal/discovery/OpenDaikinACUnitDiscoveryService.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.opendaikin.internal.discovery;
+
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.ws.rs.client.ClientBuilder;
+
+import org.eclipse.smarthome.config.discovery.AbstractDiscoveryService;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.config.discovery.DiscoveryResultBuilder;
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceCallback;
+import org.eclipse.smarthome.core.net.NetUtil;
+import org.eclipse.smarthome.core.thing.ThingUID;
+import org.openhab.binding.opendaikin.OpenDaikinBindingConstants;
+import org.openhab.binding.opendaikin.internal.OpenDaikinWebTargets;
+import org.openhab.binding.opendaikin.internal.config.OpenDaikinConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Discovery service for Daikin AC units.
+ *
+ * @author Tim Waterhouse <tim@timwaterhouse.com> - Initial contribution
+ *
+ */
+public class OpenDaikinACUnitDiscoveryService extends AbstractDiscoveryService {
```

Comment:
You can also use `@Component(` annotation here. And than remove the xml in  `OSGI-INF`: Put a `.gitignore` file in the ` OSGI-INF` directory with content `*.xml` to avoid having the generated xml by the annotations added to git.